### ISSUE TITLE
gcsfs 0.5.2 breaks on permission issue

### ISSUE THREAD

**Issue by DPGrev** [dask/gcsfs#218]
After updating from 0.5.1 to 0.5.2 we get the following error:

```
<<Project Number>>-compute@developer.gserviceaccount.com does not have serviceusage.services.use access to project <<Project Number>>.
```

This does not happen for GCFS 0.5.1.

**Comment by TomAugspurger** [comment]
Sorry about that, will get a fixed release out soon.

I noticed the same error yesterday when my default Google project didn't match the project I was passing to GCSFileSystem.

Can you confirm that yours match? It's `gcloud config get-value project`

**Comment by TomAugspurger** [comment]
I should mention, this happened when using `google_default` to connect (the default `token`, if successful, or explicitly with `token="google_default"`.

**Comment by TomAugspurger** [comment]
@DPGrev can you check with master? If it's the same issue, the call create the GCSFileSystem will now raise when the projects don't match up.

If it's a different issue, it'd be good to diagnose that before I do a release later today.

**Comment by sukhinin541** [comment]
FWIW, I get the same `does not have serviceusage.services.use access` error in 0.5.3, in particular when I try pd.read_csv(). Also, pandas_df.to_csv() just fails silently. Rolling gcsfs back to 0.5.1 resolves these issues.

**Comment by TomAugspurger** [comment]
@sukhinin541 can you share more information? In particular,

1. What auth mechanism are you using?
2. Does your `gcloud config get-value project` match the project you're using with gcsfs?

@DPGrev could you do the same?

**Comment by sukhinin541** [comment]
@TomAugspurger, here are my answers:

1. When I use gcsfs directly, I'm letting it determine my credentials, i.e. `fs = gcsfs.GCSFileSystem()`
2. `gcloud config get-value project` matches what `fs.project` returns

Also, when I call `read_csv()/to_csv()` Pandas methods, I'm only indirectly using gcsfs, so I'm not sure which auth mechanism is utilized in that scenario.

Let me know if you need more info.

**Comment by ctivanovich** [comment]
I am having the same error, also with `gcsfs-0.5.3`. I similarly have authenticated a service account in my linux environment, and have gcsfs auto-detect the environment's config. I have confirmed I don't this issue with an early version of gcsfs.

**Comment by TomAugspurger** [comment]
OK. I may have to make some changes to the requester-pays API then. I Most likely we'll need an explicit `requester_pays=False / True` flag in `GCSFileSystem.__init__`, rather than always including the `project` in the requests made to GCS.

I wish I understood a bit more about why we're getting the errors though. In my case, the `project` I passed to GCSFileSystem didn't match the project of the account I authenticated with. 

@sukhinin541 

> When I use gcsfs directly, I'm letting it determine my credentials, i.e. fs = gcsfs.GCSFileSystem()

Can you explicitly use `token="google_default"`? That's the first thing gcsfs tries, but it falls back to other auth methods.

**Comment by sukhinin541** [comment]
@TomAugspurger, when I provide `token='google_default'`, I still have the following situation:

1. `gcloud config get-value project` matches what `fs.project` returns
2. I get `does not have serviceusage.services.use access` error when I try to use `fs.open()`

Also, the project number mentioned in the does-not-have-access error message is the correct number for the project ID that's returned by `fs.project`

**Comment by TomAugspurger** [comment]
Actually, I was able to reproduce this using `GCSFileSystem.get` on an object in a bucket I own. Debugging a bit more.

**Comment by TomAugspurger** [comment]
Ah, nevermind, my user account didn't have permissions for that bucket. So it failed for GCSFS 0.4.x and master.

**Comment by TomAugspurger** [comment]
#224 was just merged, which should fix things for you all. If you could try installing from master I'd appreciate it.

I'll do a release tomorrow.

**Comment by sukhinin541** [comment]
I can confirm that the issue goes away after installing from the master.

**Comment by TomAugspurger** [comment]
Thanks for confirming. I just released gcsfs 0.6.0 with those changes. For everyone not using requester-pays, the behavior should be identical to 0.4.x

**Comment by Trollgeir** [comment]
I suddenly seem to have the same issue with any version of `gcsfs` including `0.7.1`.
```
import gcsfs
fs = gcsfs.GCSFileSystem()
fs.ls("")

OSError: Forbidden: https://www.googleapis.com/storage/v1/b/
XXX@XXX does not have serviceusage.services.use access to the Google Cloud project.
```
I have confirmed that I'm on the same project, and the weird thing is that `gsutil ls` works just fine - which afaik shares the credentials as `gcsfs` ?
